src/clj/rems/db/user_mappings.clj

```diff
@@ -0,0 +1,46 @@
+(ns rems.db.user-mappings
+  (:require [clojure.tools.logging :as log]
+            [rems.db.core :as db]
+            [rems.json :as json]
+            [medley.core :refer [deep-merge]]
+            [schema.coerce :as coerce]
+            [schema.core :as s]))
+
+(s/defschema UserMappingsDataDb
+  {:from s/Str
+   :from-value s/Str
+   :to-value s/Str})
+
+(def ^:private coerce-UserMappingsDataDb
+  (coerce/coercer! UserMappingsDataDb coerce/string-coercion-matcher))
+
+(defn- format-user-mappings [user-mappings]
+  (let [usermappingsdata (-> (json/parse-string (:usermappingsdata user-mappings))
+                             coerce-UserMappingsDataDb)
+        {:keys [from from-value to-value]} usermappingsdata]
+    {from {from-value to-value}}))
+
+;; TODO: think about cache namespace refactoring
+(def ^:private user-mappings-cache (atom nil))
+
+(defn reset-cache! []
+  (reset! user-mappings-cache nil))
+
+(defn reload-cache! []
+  (log/info :start #'reload-cache!)
+  (let [user-mappings (->> (db/get-user-mappings)
+                           (map format-user-mappings)
+                           (reduce deep-merge {}))]
```

Comment:
Here I'm thinking that we could `(index-by [:from :from-value] mappings)`.

Reply:
removed cache in favor of indexed db table (and query)